File: src/main/java/github/stephenwanjala/travelagency/dao/CustomerDAO.java
package github.stephenwanjala.travelagency.dao;

import github.stephenwanjala.travelagency.model.Customer;

import java.util.List;

public interface CustomerDAO {
    List<Customer> getAllCustomers();
    Customer getCustomerById(int id);
    void addCustomer(Customer customer);
    void updateCustomer(Customer customer);
    void deleteCustomer(int id);
}

File: src/main/kotlin/data/dao/CustomerDao.kt
package data.dao

import androidx.room.*
import models.Customer

@Dao
interface CustomerDao {
    @Query("SELECT * FROM Customer")
    suspend fun getAllCustomers(): List<Customer>

    @Query("SELECT * FROM Customer WHERE id = :id")
    suspend fun getCustomerById(id: Int): Customer

    @Upsert
    suspend fun insertCustomer(customer: Customer)

    @Update
    suspend fun updateCustomer(customer: Customer)

    @Delete
    suspend fun deleteCustomer(customer: Customer)
}